SalesforceSDK/Salesforce.SDK.Core/Source/Rest/ClientManager.cs

```diff
@@ -99,5 +99,16 @@ public RestClient GetRestClient()
             }
             return restClient;
         }
+
+        public RestClient GetUnAuthenticatedRestClient(string instanceUrl)
+        {
+            var restClient = PeekUnAuthenticatedRestClient(instanceUrl);
+            return restClient;
+        }
+
+        public RestClient PeekUnAuthenticatedRestClient(string instanceUrl)
```

Comment:
Looks like ```GetUnAuthenticatedRestClient``` and ```PeekUnAuthenticatedRestClient``` are doing the same thing. If that's the case, we don't need ```PeekUnAuthenticatedRestClient```. On ```Android```, we have both because one is synchronous and the other is asynchronous and has a callback mechanism.